### PR TITLE
CommandLineInterface: add storage-layout option to --combined-json

### PR DESCRIPTION
**Pull request by MrChico**
Superseeds #8525 which github won't let me reopen :(

### CODE REVIEW

File: solc/CommandLineInterface.cpp [hunk 1]
```diff
@@ -433,6 +438,18 @@ void CommandLineInterface::handleABI(string const& _contract)
...
 }
 
+void CommandLineInterface::handleStorageLayout(string const& _contract)
+{
+	if (!m_args.count(g_argStorageLayout))
+		return;
+
+	string data = jsonCompactPrint(m_compiler->storageLayout(_contract));
+	if (m_args.count(g_argOutputDir))
+		createFile(m_compiler->filesystemFriendlyName(_contract) + "._storage.json", data);
```
Comment (chriseth) [inline solc/CommandLineInterface.cpp]: Why `._storage`?

File: solc/CommandLineInterface.cpp [hunk 2]
```diff
@@ -1653,6 +1673,7 @@ void CommandLineInterface::outputCompilationResults()
 		handleSignatureHashes(contract);
 		handleMetadata(contract);
 		handleABI(contract);
+ 		handleStorageLayout(contract);
```
Comment (chriseth) [inline solc/CommandLineInterface.cpp]: Please use tabs.

File: solc/CommandLineInterface.cpp (hunk 1 above)
Comment (MrChico) [reply solc/CommandLineInterface.cpp]:
```suggestion
		createFile(m_compiler->filesystemFriendlyName(_contract) + "_storage.json", data);
```

Comment (chriseth) [reply solc/CommandLineInterface.cpp]: just do it!

File: solc/CommandLineInterface.cpp (hunk 2 above)
Comment (MrChico) [inline solc/CommandLineInterface.cpp]:
```suggestion
		handleStorageLayout(contract);
```

**Comment by chriseth** [comment]
Please add a changelog entry and squash.

**Comment by MrChico** [comment]
Okk

File: Changelog.md [hunk 3]
```diff
@@ -5,7 +5,7 @@ Language Features:
 
 Compiler Features:
  * Metadata: Added support for IPFS hashes of large files that need to be split in multiple chunks.
-
+ * Commandline Interface: Enable output of storage layout with `--storage-layout`,
```
Comment (leonardoalt) [inline Changelog.md]:
```suggestion
 * Commandline Interface: Enable output of storage layout with `--storage-layout`.
```

**Review by leonardoalt — COMMENTED** [review]
Looks good, just this one typo

File: Changelog.md (hunk 3 above)
Comment (MrChico) [reply Changelog.md]: fixed

**Review by chriseth — APPROVED** [review]

**Pull request merged**